operator/controller/src/test/java/io/apicurio/registry/operator/it/SmokeITTest.java

```diff
@@ -148,4 +166,104 @@ void testIngress() {
             return true;
         });
     }
+
+    @Test
+    void testEmptyHostDisablesIngress() {
+        // spotless:off
+        var registry = new ApicurioRegistry3();
+        var meta = new ObjectMeta();
+        meta.setName("demo");
+        meta.setNamespace(namespace);
+        registry.setMetadata(meta);
+        registry.setSpec(ApicurioRegistry3Spec.builder()
+                .app(ApicurioRegistry3SpecApp.builder()
+                        .host(ingressManager.getIngressHost("demo-app"))
+                        .build())
+                .ui(ApicurioRegistry3SpecUI.builder()
+                        .host(ingressManager.getIngressHost("demo-ui"))
+                        .build())
+                .build());
+        // spotless:on
+
+        client.resource(registry).create();
+
+        // Wait for Ingresses
+        await().untilAsserted(() -> {
+            assertThat(client.network().v1().ingresses().inNamespace(namespace).withName("demo-app-ingress")
+                    .get()).isNotNull();
+            assertThat(client.network().v1().ingresses().inNamespace(namespace).withName("demo-ui-ingress")
+                    .get()).isNotNull();
+        });
+
+        // Check that REGISTRY_API_URL is set
+        var uiDeployment = client.apps().deployments().inNamespace(namespace).withName("demo-ui-deployment")
+                .get();
+        verify_REGISTRY_API_URL_isSet(registry, uiDeployment);
+
+        // Disable host and therefore Ingress
+        registry.getSpec().getApp().setHost("");
+        registry.getSpec().getUi().setHost("");
+
+        // The remote test does not work properly. As a workaround the CR will be deleted and recreated
```

Comment:
I'm fine with the workaround for now, I would add `// TODO: ` so that the IDE will highlight it, but it's a detail.